odo/backends/dask_array.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+from datashape import discover
+from toolz import merge, accumulate
+from datashape.dispatch import dispatch
+from datashape import DataShape
+from operator import add
+import itertools
+
+from dask.array.core import rec_concatenate, Array, getem, get, names, from_array
+from dask.core import flatten
+from dask.compatibility import long
+
+from ..append import append
+from ..convert import convert
+from ..resource import resource
+
+@discover.register(Array)
+def discover_dask_array(a, **kwargs):
+    block = a._get_block(*([0] * a.ndim))
+    return DataShape(*(a.shape + (discover(block).measure,)))
```

Comment:
We not implement a `dtype` property on dask.arrays so this could be simplified.  The `datashape.from_numpy` function can be useful here.